### PR TITLE
[Ceph-Volume] [Tier-2] Check LVM tags post removing OSD

### PR DESCRIPTION
**Pull request by pranavprakash20**
Polarion Tc's included : CEPH-83575103    Bug 2067952 -Check LVM tags post removing OSD (-zap)

Run Log:  http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-7F3O7P/

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: conf/pacific/ceph_volume/tier-2_cephvolume_5node_.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+globals:
```
Comment (vamahaja) [inline conf/pacific/ceph_volume/tier-2_cephvolume_5node_.yaml:1]: Add test suite for `quincy` as well

File: tests/ceph_volume/test_ceph_volume_lvm_tags_after_osd_rm.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        osd_node, format="json", expr="name=osd", sudo=True
+    )
+    container_ids = [item["Names"][0] for item in loads(running_containers)]
+    lvm_list = get_lvm_list(container_ids[0], osd_node)
+
+    # Identify an OSD ID to perform
+    osd_id = list(lvm_list.keys())[0]
+
+    # Change the replication to 2 / default is 3
+    out = CephAdm(osd_node).ceph.osd(command="pool ls detail", format="json")
```
Comment (vamahaja) [inline tests/ceph_volume/test_ceph_volume_lvm_tags_after_osd_rm.py:33]: Is this any general command or specific commands ?

File: tests/ceph_volume/test_ceph_volume_lvm_tags_after_osd_rm.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    Args:
+        **kw: Key/value pairs of configuration information to be used in the test.
+    """
+    osd_node = ceph_cluster.get_nodes(role="osd")[0]
+
+    # Get the lvm list before osd remove
+    running_containers, out = get_running_containers(
+        osd_node, format="json", expr="name=osd", sudo=True
+    )
+    container_ids = [item["Names"][0] for item in loads(running_containers)]
```
Comment (vamahaja) [inline tests/ceph_volume/test_ceph_volume_lvm_tags_after_osd_rm.py:26]: use `dict.get` in all places

File: tests/ceph_volume/test_ceph_volume_lvm_tags_after_osd_rm.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        if "No OSD remove/replace operations reported" in out:
+            break
+    if w.expired:
+        raise FailedToRemoveOsdError("Failed to perform osd rm operation. Timed out!")
+
+    # Get osd tree and check whether the deleted osd is present or not
+    out = CephAdm(osd_node).ceph.osd(command="tree", format="json")
+    osd_tree = loads(out[0])
+    for osd_ in osd_tree["nodes"]:
+        if osd_["id"] == osd_id:
```
Comment (vamahaja) [inline tests/ceph_volume/test_ceph_volume_lvm_tags_after_osd_rm.py:65]: both conditions can be combined with `and`

File: cli/utilities/utils.py [hunk 5]
```diff
@@ -437,3 +438,18 @@ def is_node_online(node):
...
+    """
+    Returns the lvm list for the given container id
+    Args:
+        container_id (str): ID of the container
+        node (ceph): Node to execute the cmd
+
+    Returns (str): output of the command
+    """
+    cmd = f"podman exec {container_id} ceph-volume lvm list --format {format}"
+    out, _ = node.exec_command(cmd=cmd, sudo=True)
```
Comment (vamahaja) [inline cli/utilities/utils.py:453]: Use function - https://github.com/red-hat-storage/cephci/blob/master/cli/utilities/utils.py#L76

File: cli/utilities/utils.py [hunk 6]
```diff
@@ -437,3 +438,18 @@ def is_node_online(node):
         return True
     log.error(f"{node.hostname} is DOWN, Ping failed")
     return False
+
+
+def get_lvm_list(container_id, node, format="json"):
```
Comment (vamahaja) [inline cli/utilities/utils.py:443]: change name to `get_lvm_on_osd_container`

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: conf/pacific/ceph_volume/tier-2_cephvolume_5node_.yaml (hunk 1 above)
Comment (pranavprakash20) [reply conf/pacific/ceph_volume/tier-2_cephvolume_5node_.yaml:1]: added.

File: tests/ceph_volume/test_ceph_volume_lvm_tags_after_osd_rm.py (hunk 2 above)
Comment (pranavprakash20) [reply tests/ceph_volume/test_ceph_volume_lvm_tags_after_osd_rm.py:30]: It is an osd specific command Vaibhav

File: tests/ceph_volume/test_ceph_volume_lvm_tags_after_osd_rm.py (hunk 4 above)
Comment (pranavprakash20) [reply tests/ceph_volume/test_ceph_volume_lvm_tags_after_osd_rm.py]: Ack. Done

File: cli/utilities/utils.py (hunk 6 above)
Comment (pranavprakash20) [reply cli/utilities/utils.py]: Ack. changed

File: suites/pacific/ceph_volume/tier-2_ceph_volume.yaml [hunk 7]
```diff
@@ -0,0 +1,83 @@
+#===============================================================================================
```
Comment (vamahaja) [inline suites/pacific/ceph_volume/tier-2_ceph_volume.yaml:1]: Lets change the test suite description. Add below details - 

- Test Suite - Description
- Setup - Test cases which are part of cluster setup
- Test Cases - List of test cases 

Refer - https://github.com/red-hat-storage/cephci/pull/2382/files#diff-ac5017c9a9c4124edf8ca6cdbc4ddb6f437af3a0db36126e544e5be473f6fbd1

**Review by vamahaja — CHANGES_REQUESTED** [review]

**Comment by pranavprakash20** [comment]
Latest run log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-GG77IP/

File: cli/ceph/orch/osd.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+            osd_id (str): ID of the osd to remove
+            status (bool): Whether to check for the rm status or not
+            format (bool): Format in which status has to be fetched
+            kw (dict): Key/value pairs that needs to be provided to the installer
+
+            Supported Keys:
+                force (bool): Whether to append force with rm command
+                --zap (bool) : Whether to append zap with rm command
+        """
+        if status:
```
Comment (vamahaja) [inline cli/ceph/orch/osd.py:26]: Better way -
```
        cmd = [self.base_cmd, "rm"]
        if status:
                cmd.append("status")

        elif osd_id:
                cmd.append(osd_id)

        cmd.append(build_cmd_from_args(**kw))
        out = self.execute(sudo=True, cmd=cmd.join(" "))
```

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: cli/ceph/orch/osd.py (hunk 8 above)
Comment (pranavprakash20) [reply cli/ceph/orch/osd.py:27]: Noted, done

**Review by vamahaja — APPROVED** [review]

**Pull request merged**